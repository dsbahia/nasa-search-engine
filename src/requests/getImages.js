import axios from "axios";

const getImages = async (query) => {
  if(!query) {
    return [];
  }

  try {
    const response = await axios.get(`https://images-api.nasa.gov/search?q=${query}`);
    const imageResults = response.data.collection.items;
    const parsedImages = imageResults.filter(
      (imageResult) => imageResult.data[0].media_type === "image"
    );
    const images = parsedImages.map((image) => image.links[0].href);
    return images;
  } catch (error) {
    console.error(error);
    throw error;
  } 
}

export default getImages;
