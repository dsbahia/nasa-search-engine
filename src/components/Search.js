import React, { useState } from "react";
import "../styles/search.css";
import getImages from "../requests/getImages";

const Search = ({ setSearchResults }) => {
  const [query, setQuery] = useState("");

  const handleSubmit = async (event) => {
    event.preventDefault();
    const images = await getImages(query);
    setSearchResults(images);
  };

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          className="search"
          placeholder="Enter keywords..."
          value={query}
          onChange={(e) => setQuery(e.target.value)}
        />
        <button className="search-button" type="submit">
          Go!
        </button>
      </form>
    </div>
  );
};

export default Search;
