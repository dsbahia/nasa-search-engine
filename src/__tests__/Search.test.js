import React from "react";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import Search from "../components/Search";
import * as getImages from "../requests/getImages";

describe("Search", () => {
  it("Renders Correctly", () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Updates input value when typing", () => {
    render(<Search />);
    const inputElement = screen.getByPlaceholderText("Enter keywords...");

    fireEvent.change(inputElement, {target: { value: "space" }} );
    
    expect(inputElement.value).toBe("space");
  });

  it("Calls setSearchResults when form is submitted", async () => {
    const setSearchResultsMock = jest.fn();
    const mockReturn = jest.fn();
    jest.spyOn(getImages, "default").mockResolvedValue(mockReturn);

    render(<Search setSearchResults={setSearchResultsMock} />);
    const inputElement = screen.getByPlaceholderText("Enter keywords...");
    const submitButton = screen.getByText("Go!");

    const input = "moon"
    fireEvent.change(inputElement, { target: { value: input } });
    fireEvent.click(submitButton);

    await waitFor(() => {
      expect(getImages.default).toBeCalledWith(input)
    });

    await waitFor(() => {
      expect(setSearchResultsMock).toHaveBeenCalledWith(mockReturn);
    });
  });
});
