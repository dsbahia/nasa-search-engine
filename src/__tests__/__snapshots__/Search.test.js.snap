// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Renders Correctly 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        class="search"
        placeholder="Enter keywords..."
        type="text"
        value=""
      />
      <button
        class="search-button"
        type="submit"
      >
        Go!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
